multimodal/src/autogluon/multimodal/utils/export.py

```diff
@@ -153,49 +153,61 @@ def export_onnx(
+        Optimize the predictor's model for inference.
+
+        Under the hood, the implementation would convert the PyTorch module into an ONNX module, so that
+        we can leverage efficient execution providers in onnxruntime for faster inference.
 
         Parameters
         ----------
         data
             Raw data used to trace and export the model.
             If this is None, will check if a processed batch is provided.
-        path
-            The export path of onnx model.
-        batch_size
-            The batch_size of export model's input.
-            Normally the batch_size is a dynamic axis, so we could use a small value for faster export.
+        providers : dict or str, default=None
+            A list of execution providers for model prediction in onnxruntime.
+
+            By default, the providers argument is None. The method would generate an ONNX module that
+            would perform model inference with TensorrtExecutionProvider in onnxruntime, if tensorrt
+            package is properly installed. Otherwise, the onnxruntime would fallback to use CUDA or CPU
+            execution providers instead.
 
         Returns
         -------
-        trt_module : OnnxModule
+        onnx_module : OnnxModule
             The onnx-based module that can be used to replace predictor._model for model inference.
         """
-        import onnx
-        import torch
+        data_dict = {}
+        for col_name, col_type in self._column_types.items():
+            if col_type in [NUMERICAL, CATEGORICAL, NULL]:
+                data_dict[col_name] = [0, 1]
+            elif col_type == TEXT:
+                data_dict[col_name] = ["some text", "some other text"]
+            elif col_type in [IMAGE_PATH]:
+                data_dict[col_name] = ["/not-exist-dir/xxx.jpg", "/not-exist-dir/yyy.jpg"]
+            else:
+                raise ValueError(f"unsupported column type: {col_type}")
+        data = pd.DataFrame.from_dict(data_dict)
 
-        from .onnx import OnnxModule
+        onnx_path = self.export_onnx(data=data, path=self.path, truncate_long_and_double=True)
+        onnx_module = OnnxModule(onnx_path, providers)
+        onnx_module.input_keys = self._model.input_keys
+        onnx_module.prefix = self._model.prefix
+        onnx_module.get_output_dict = self._model.get_output_dict
 
-        truncate_long_and_double = False
-        onnx_path = self.export_onnx(
-            data=data, path=path, batch_size=batch_size, truncate_long_and_double=truncate_long_and_double
-        )
+        # To use the TensorRT module for prediction, simply replace the _model in the predictor
+        self._model = onnx_module
 
-        logger.info("Loading ONNX file from path {}...".format(onnx_path))
-        onnx_model = onnx.load(onnx_path)
+        # Evaluate and cache TensorRT engine files
+        logger.info("Compiling ... (this may take a few minutes)")
+        _ = self.predict(data)
```

Comment:
Is this compilation only for TensorRT? Does onnx also use it?

Reply:
Yes, this is only for TensorRT.